manifests/gcp_marketplace/schema.yaml

```diff
@@ -114,6 +114,11 @@ properties:
     type: string
     x-google-marketplace:
       type: NAMESPACE
+  needDefaultBucket:
```

Comment:
@numerology I think it'll be better UX to disable default bucket provisioning when the `managedstorage.gcsBucketName` field is supplied by users.

The supplied gcsBucketName will be used for default bucket too, so that in all cases, default bucket will be available to KFP.

Reply:
> The supplied gcsBucketName will be used for default bucket too

Just for my education, will the bucket provided by `managedstorage.gcsBucketName` accessible by the magic placeholder?

Reply:
Yes, it was my hacky solution, because I couldn't figure out how to read schema.yaml from the init_script. If possible, we should remove this branch logic and let init_script generate the configmap for this bucket.
https://github.com/kubeflow/pipelines/blob/03904ebacd005a9b4bbb7047b130da2a7a50fec6/manifests/gcp_marketplace/chart/kubeflow-pipelines/templates/pipeline.yaml#L708-L724

Reply:
Perhaps we should just use the non-emptiness of `gcsBucketName` to decide whether to generate a default bucket.
WDYT?
You can also use it in the conditional expressions.

Reply:
Yep that's doable as well, without introducing too much dependencies/complexity.

I've dropped the ball on this PR for a while, I'll update it shortly.